sdks/java/extensions/sql/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -280,4 +285,48 @@ Schema.FieldType SimpleType() :
     }
 }
 
+SqlSetOptionBeam SqlSetOptionBeam(Span s, String scope) :
+{
+    SqlIdentifier name;
+    final SqlNode val;
+}
+{
+    (
+        <SET> {
```

Comment:
SET/UNSET should be supported already, can you verify that?

Reply:
I mean the parser supports SET/UNSET(RESET?) syntax, although it's ignored in Beam implementation.

Reply:
The parser does support SET/UNSET/RESET, however it can not execute or plan the resulting `SqlSetOption`. This is  a copy and paste of the calcite parser config with `SqlSetOption` replaced with `SqlSetOptionBeam`.

Reply:
I suppose this is about CLI part, and `BeamSqlCli` is the main point. --Pls correct me if any mis-understand since I see the test code uses `JdbcDriver`. In the prior way, you could extend  `execute` method to support `SqlSetOption` directly.

Ps, my CLI(a bash script which invokes BeamSqlCli) looks like:
```
CLI= BeamSqlCli.new
loop:
  EXPLAIN ...
  | (INSERT... ) SELECT ...
  | CREATE TABLE ...
  | SET/UNSET ...
  | EXIT
```

Reply:
The JDBC path gives us no way to extend execute. When accessing Beam SQL through that path all outputs from the parser must be self executing, which `SqlSetOptionBeam` is. Are you already using `SET/RESET` for something?